client/client.go

```
@@ -123,6 +143,29 @@ func (c *Client) doRequest(req *http.Request) ([]byte, int, error) {
 	return body, res.StatusCode, err
 }
 
+func (c *Client) authenticate() error {
+	mapping, encodedData := map[string]string{"password": c.AdminPassword}, new(bytes.Buffer)
+	json.NewEncoder(encodedData).Encode(mapping)
```

i think this error is not handled